src/mats_l1_processing/L1_calibration_functions.py

```diff
@@ -779,15 +779,17 @@ def desmear(image, nrskip, exptimeratio, fill=None):
     """
 
     nrow, ncol = image.shape
-    nr = nrow+nrskip
+    nr = nrow-nrskip
     weights = np.tril(
-        exptimeratio*np.ones([nr, nr]), -(nrskip+1))+np.diag(np.ones([nr]))
+        exptimeratio*np.ones([nrow, nrow]), -(nrskip+1))+np.diag(np.ones([nrow]))
     if nrskip > 0:
-        extimage = np.vstack((fill, image)) 
+        extimage = image - \
+            np.tril(exptimeratio*np.ones([nrow, nrow]), -
+                    1) @ np.vstack((fill, np.zeros([nr, ncol])))
```

Comment:
This row break is a bit unfortunate... (In other news, both `@` and `np.tril` are new to me! Is  `@` shorthand for matrix multiply or something?)

Reply:
yes I did not like it either -  autoformater 

@ is matrix multipy. -I like it  A@B is much easier than np.mulmat(A,B) 
tirl is a lower triangular matrix with an offset of inthis case -1